internals/overlord/servstate/manager.go

```diff
@@ -237,6 +237,12 @@ func (m *ServiceManager) acquirePlan() (release func(), err error) {
 	return release, nil
 }
 
+// DryStart implements StateManager.DryStart.
+func (m *ServiceManager) DryStart() error {
+	_, err := plan.ReadDir(m.pebbleDir)
```

Comment:
If the intention is that ```DryStart()``` should be the place where early unwanted-side-effect-free init happens (which is what I understand), I think we can use this is an example:

```
        releasePlan, err := m.acquirePlan()
        if err != nil {
                return err
        }
        defer releasePlan()
```

This will initialize ```m.plan```, which will prevent any future acquire from re-doing it.